### ISSUE TITLE
wvunpack should ignore qmode from WavpackGetMode() for WAV family output

### ISSUE THREAD

**Issue by nu774** [dbry/WavPack#13]
As the title says.
When a big endian caf file is encoded with -r option set, wvunpack decodes it into a big endian wav file by default.

**Comment by dbry** [comment]
Thanks, good catch!

There are actually several things along those lines that I need to look into. Without even trying, I'm afraid that using the --skip and --until options might produce some wrong results with other formats/endians too.

And I really need to figure out how I want the -r (raw) option in wvunpack to work. I think it should always output the native (original) endian, but then I should have alternate versions like --raw-be and --raw-le to force a specific format.

But yeah, WAVs should always be LE...  :-)

**Comment by dbry** [comment]
I decided that to really fix this (and related) issues correctly I need to store the source file format in the file, along with the original extension, and create an array of available output formats that wvunpack can use.

So now, if you pack a big-endian caf file with -r and unpack it, you'll get a big-endian caf file (with a freshly generated generic caf header). If you want a wav file instead, the -w option will do it, and it will not be big-endian (obviously). I think I covered all the bases here, but feel free to point out if I missed anything.

Thanks again!
